nexus/reconfigurator/planning/src/example.rs

```diff
@@ -34,17 +33,100 @@ pub struct ExampleSystem {
     pub(crate) sled_rng: TypedUuidRng<SledKind>,
 }
 
-impl ExampleSystem {
-    pub fn new(
-        log: &slog::Logger,
-        test_name: &str,
-        nsleds: usize,
-    ) -> ExampleSystem {
+/// Returns a collection, planning input, and blueprint describing a pretty
+/// simple system.
+///
+/// The test name is used as the RNG seed.
+pub fn example(
+    log: &slog::Logger,
+    test_name: &str,
+) -> (Collection, PlanningInput, Blueprint) {
+    let (example, blueprint) =
+        ExampleSystemBuilder::new(log, test_name).build();
+    (example.collection, example.input, blueprint)
+}
+
+/// A builder for the example system.
+#[derive(Debug, Clone)]
+pub struct ExampleSystemBuilder {
+    log: slog::Logger,
+    test_name: String,
+    nsleds: usize,
+    no_zones: bool,
```

Comment:
Nit: I think negative flags can get confusing and usually prefer to see positive flags so we don't have to use `!no_..` in conditions. How about changing these to `use_zones`, `add_zones`, or `create_zones`, etc...

Reply:
Yeah, that makes sense. Will do.

Reply:
Done.